syntax = "proto3";

import "google/protobuf/empty.proto";

option go_package = "api/";

message MediaItemResultRequest {
    string id = 1;
    string status = 2;
    optional string filename = 3;
    optional string mimeType = 4;
    optional string sourceUrl = 5;
    optional string previewUrl = 6;
    optional string thumbnailUrl = 7;
    optional string type = 8;
    optional int32 width = 9;
    optional int32 height = 10;
    optional string creationTime = 11;
    optional string cameraMake = 12;
    optional string cameraModel = 13;
    optional string focalLength = 14;
    optional string apertureFNumber = 15;
    optional string isoEquivalent = 16;
    optional string exposureTime = 17;
    optional string fps = 18;
    optional double latitude = 19;
    optional double longitude = 20;
}

message MediaItemPlaceRequest {
    string id = 1;
    string postcode = 2;
    string country = 3;
    string state = 4;
    string district = 5;
    string county = 6;
    string city = 7;
    string town = 8;
    string suburb = 9;
    string road = 10;
}

service APIService {
    rpc SaveMediaItemResult(MediaItemResultRequest) returns (google.protobuf.Empty);
    rpc SaveMediaItemPlace(MediaItemPlaceRequest) returns (google.protobuf.Empty);
}
